content/client-lib-development-guide/features.textile

```diff
@@ -367,6 +367,9 @@ h3(#realtime-connection). Connection
 ** @(RTN19a)@ Any @ProtocolMessage@ that is awaiting an @ACK@/@NACK@ on the old transport will not receive the @ACK@/@NACK@ on the new transport. The client library must therefore resend any @ProtocolMessage@ that is awaiting a @ACK@/@NACK@ to Ably in order to receive the expected @ACK@/@NACK@ for that message. The Ably service is responsible for keeping track of messages, ignoring duplicates and responding with suitable @ACK@/@NACK@ messages
 ** @(RTN19b)@ If there are any pending channels i.e. in the @ATTACHING@ or @DETACHING@ state, the respective @ATTACH@ or @DETACH@ message should be resent to Ably
 ** @(RTN19b)@ If a @SYNC@ is underway, ensure the client library adheres to @RTP3@
+* @(RTN20)@ Heartbeats
+** @(RTN20a)@ If a transport does not receive a heartbeat for a period greater than the sum of the @heartbeatPeriod@ (sent in the @connectionDetails@ in the @CONNECTED@ message received on that transport) and the @realtimeRequestTimeout@, that transport may be disconnected. This requirement is optional; in deciding whether to implement, client library developers should take into account whether the transport in question is susceptible to undetected dropped connections.
```

Comment:
> Buffering in a cellular connection can be 6s.

FWIW an approximately constant buffer would be fine no matter how big, since the timer is since the last heartbeat (or the `CONNECTED`); it'd only be a problem if it increased from 0 to >10s between heartbeats.

Also, at the risk of stating the obvious, we already use `realtimeRequestTimeout` for a bunch of stuff in the client libs. If we believe 10s is too tight for heartbeats, doesn't that imply it's also too tight for all that stuff? (Especially since most of the things it's already used for involve a round-trip, unlike heartbeats which are one-way). Which would imply we need to increase `realtimeRequestTimeout`